steps/optimizers.py

```diff
@@ -106,9 +122,9 @@ def optimize_variational_circuit(
         constraints_cost_function_specs["backend"] = backend
         constraints_cost_function_specs["target_operator"] = constraint_op
         constraint_cost_function = create_object(constraints_cost_function_specs)
-        constraint_cost_function_wrapper = (
-            lambda params: constraint_cost_function.evaluate(params).value
-        )
+        constraint_cost_function_wrapper = lambda params: constraint_cost_function.evaluate(
+            params
+        ).value
```

Comment:
Probably late to the party as the branch was already merged, but I'd like to add my two cents for the sake of similar scenarios in the future 🙂.

Wouldn't be better to use an explicit local function instead of a lambda assigned to a named variable? [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) actually recommends that, with the reasoning being:
>This is more useful for tracebacks and string representations in general. The use of the assignment statement eliminates the sole benefit a lambda expression can offer over an explicit def statement (i.e. that it can be embedded inside a larger expression).

This is how it would look like:

```python
...
constraint_cost_function = create_object(constraints_cost_function_specs)
def constraint_cost_function_wrapper(params):
    return constraint_cost_function.evaluate(params).value
```